typhos/display.py

```diff
@@ -1447,19 +1441,20 @@ def search_for_templates(self):
                 logger.debug('Adding macro template %s: %s (total=%d)',
                              display_type, template, len(template_list))
 
-            # 2. Composite heuristics, if enabled
-            if self._composite_heuristics and view == 'detailed':
-                if self.suggest_composite_screen(cls):
-                    template_list.append(DETAILED_TREE_TEMPLATE)
-
-            # 3. Templates based on class hierarchy names
+            # 2. Templates based on class hierarchy names
             filenames = utils.find_templates_for_class(cls, view, paths)
             for filename in filenames:
                 if filename not in template_list:
                     template_list.append(filename)
                     logger.debug('Found new template %s: %s (total=%d)',
                                  display_type, filename, len(template_list))
 
+            # 3. Ensure that the detailed tree template makes its way in for
+            #    all top-level screens, if no class-specific screen exists
```

Comment:
This is an interesting choice.  It's useful but has added to my confusion about template names.  Why do we do this?

Reply:
So it's like:
* "AT2L0 is a complex device. If it has a custom screen when loaded, use it"
* In a scenario where AT2L0 didn't have a custom screen, we'd want it to use the detailed tree as the "next best thing"

It's all confusing, for sure. Does that touch on your confusion on this aspect or is it something else?

Reply:
It's more that `detailed_tree.ui` shows up in each of the embedded/detailed/engineering dropdowns.

Reply:
I think that may have been unintentional